packages/pods-core/src/builtin/GDocPod.ts

```diff
@@ -0,0 +1,273 @@
+   */
+  getCommentsFromDoc = async(opts: Partial<GDocImportPodConfig>, 
+    response: Partial<NoteProps>) : Promise<Partial<NoteProps>> => {
+
+    let comments;
+    const {documentId, token, importComments } = opts;
+    const headers = {
+      Authorization: `Bearer ${token}`,
+      "Content-Type": "application/json",
+    };
+    try {
+    comments = await axios.get(`https://www.googleapis.com/drive/v2/files/${documentId}/comments`,
+    {headers});
+    
+    const items= comments.data.items;
+    if(items.length > 0) {
+    comments = items.map((item: any) => {
+      let replies;
+      if(item.replies.length > 0){
+       replies = item.replies.map((reply: any) => {
+        reply = {
+          author: reply.author.displayName,
+          content: reply.content
+        };
+        return reply;
+       })
+      }
+       item = {
+          author: item.author.displayName,
+          content: item.content,
+          replies
+        };
+        return item;
+      })
+      if(importComments?.format === "text")
+      {
+        comments = this.prettyComment(comments);
+      }
+      else {
+        comments= JSON.stringify(comments)
+      }
+       response.body = response.body?.concat(`### Comments\n\n ${comments}`)
+    }
+  }catch(error) {
+    this.L.error({
+      msg: "failed to import the comments",
+      payload: stringifyError(error),
+    });
+    throw new DendronError({ message: stringifyError(error) });
+  }
+  return response;
+  }
+  
+  /*
+   * method to prettify comment if format is text
+   */
+  prettyComment = (comments: any) => {
+    let text: string = "";
+    comments.forEach((comment: any) => {
+       text += `- ${comment.author}:  ${comment.content}\n`
+       if(comment.replies?.length> 0){
+         comment.replies.forEach((reply: any) => {
+          text += `\t - ${reply.author}: ${reply.content}\n`
+
+         })
+       }
+    })
+    return text;
+  }
+
+  async _docs2Notes(
+    entry: Partial<NoteProps>,
+    opts: Pick<ImportPodConfig, "fnameAsId"> & {
+      vault: DVault;
+    }
+  ) {
+    const { vault } = opts;
+    if (!entry.fname) {
+      throw Error("fname not defined");
+    }
+    const fname = entry.fname;
+    if(opts.fnameAsId) {
+      entry.id = fname
+    }
+    const note= NoteUtils.create({ ...entry, fname, vault });
+    return note;
+  }
+
+  createNote= async(
+    note: NoteProps,
+    engine: DEngineClient,
+    wsRoot: string,
+    vault: DVault
+  ) => {
+      const n = NoteUtils.getNoteByFnameV5({
+      fname: note.fname,
+      notes: engine.notes,
+      vault,
+      wsRoot,
+    });
+    if (!_.isUndefined(n)) {
+      if(n.custom.revisionId && n.custom.revisionId !== note.custom.revisionId) {
+        n.custom.revisionId = note.custom.revisionId;
+        n.body = note.body;
+        await engine.writeNote(n, { newNode: true });
+        }   
+       }
+    else {
+      await engine.writeNote(note, { newNode: true });
```

Comment:
I wonder if we should have a `confirmOverwrite` option so that users can have an option to get prompted on whether or not they want to overwrite their existing doc? Maybe in the next iteration.

Reply:
that would be great. @Harshita-mindfire can you add this to the [[Next|handbook.sop.weekly-journal#next]] section of your weekly journal so we can triage it?